src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/IProjectHotReloadUpdateApplier.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.HotReload.Components.DeltaApplier;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.HotReload
+{
+    /// <summary>
+    /// Provides a mechanism to apply custom updates to the running hot reload sessions. Assumes all updates can be applied via the 
+    /// IDeltaApplier created for this session
+    /// </summary>
+    [ProjectSystemContract(ProjectSystemContractScope.UnconfiguredProject, ProjectSystemContractProvider.System, Cardinality = Composition.ImportCardinality.ExactlyOne)]
+    public interface IProjectHotReloadUpdateApplier
+    {
+        /// <summary>
+        /// Returns true if there is at least one hot reload session active in the project
+        /// </summary>
+        bool HasActiveHotReloadSessions { get; }
```

Comment:
Is the idea that a caller would test this property before calling `ApplyHotReloadUpdateAsync`?

Would a single `Try*` method be more consistent with `IProjectHotReloadSessionManager.TryCreatePendingSessionAsync`?

Reply:
This check could be useful if the caller needs to do expensive work before calling `ApplyHotReloadUpdateAsync`.

But do we really need this at all?

Reply:
We do. It is the only mechanism to know whether there are active hot reload sessions and in the case of scoped CSS  files whether a design time build is needed to generate an updated css file. And even in the general case of tracking say static css files. We only want to process and send updates if the application is actually running